### ISSUE TITLE
svmradial rfe error -- predict yielding NAs

### ISSUE THREAD

**Issue by KomodoMatta** [topepo/caret#1042]
I am facing perhaps a similar issue as the person faced here three years ago: https://github.com/topepo/caret/issues/354 and also perhaps here: https://stackoverflow.com/questions/17846534/support-vector-machine-train-caret-error-kernlab-class-probability-calculations
But nothing I am trying seems to resolve this issue.

The error message I receive is:
```
Warning message:
In method$predict(modelFit = modelFit, newdata = newdata, submodels = param) :
  kernlab class prediction calculations failed; returning NAs
```

Unfortunately I do not have much programming experience, so I will preface with that. Please bear with me.

### Context:

I am running a battery of machine learnine models, which include (in addition to SVM) Artificial Neural Network, Adaboost, Elasticnet, and PLSDA. This code has worked fine for the others, so I am not sure what to change for svm. 
My Y (level) is Case or Control status. 

First, I ran a set of saturated models, and then I used `rfe` to run the same models but sparse. 

So I start with my original training data partitioned 85/15 train/test:

```
intrain <- createDataPartition(y=Y,p=0.85,list=FALSE)

training <- Endomet_Data_tr[intrain,]
testing <- Endomet_Data_tr[-intrain,]
```
Everything worked fine here for the saturated models, so then I will go into the rfe models.

#### My dataset:
```
training2 <- arrange(training, desc(Y))
training2$Y <- factor(training2$Y, levels =c("Control","Cases"))
training2$Y <- relevel(training2$Y, "Control")

set.seed(10)
rctrl1 <- rfeControl(method = "cv", number = 10, returnResamp = "all", functions = caretFuncs)

set.seed(111)
fitControl <- trainControl(method = "cv", # 
                             returnResamp = "all",
                             number = 10, 
                             classProbs = TRUE, 
                             summaryFunction = twoClassSummary)
```

I am running a 10x repeated cross validation for my battery of models. 

#### Problematic code:
```
svmrfe <- rfe(x=as.data.frame(training2[,-1]), y=factor(training2$Y),
              sizes = c(2,5,10,15,20,40),
              method = "svmRadial", 
              trControl = fitControl,
              tuneGrid = data.frame(C=10,sigma=0.001),
              rfeControl = rctrl1)

svmfinal <-svmrfe$fit
```
**It is here where I get the error:**
```
svmPredict.s <- predict(svmfinal, newdata=testing, type="raw")
```

Here is a code (for Artifical Neural Network) that ran just fine (it is essentially identical):

```nnetrfe <- rfe(x=as.data.frame(training2[,-1]), y=factor(training2$Y),
               sizes = c(2,5,10,15,20,40),
               method = "nnet",
               trControl = fitControl,
               tuneGrid = data.frame(size=1,decay=0.8),
               rfeControl = rctrl1)```


Thank you in advance for your help.

### Session Info:
```{R}
>sessionInfo()
R version 3.5.0 (2018-04-23)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

Matrix products: default

locale:
[1] LC_COLLATE=French_France.1252  LC_CTYPE=French_France.1252    LC_MONETARY=French_France.1252
[4] LC_NUMERIC=C                   LC_TIME=French_France.1252    

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] bindrcpp_0.2.2       rminer_1.4.2         ISLR_1.2             tree_1.0-39         
 [5] xgboost_0.71.2       fastAdaboost_1.0.0   boot_1.3-20          NeuralNetTools_1.5.2
 [9] neuralnet_1.33       ROCR_1.0-7           gplots_3.0.1         pROC_1.13.0         
[13] nnet_7.3-12          wesanderson_0.3.6    stargazer_5.2.2      sjstats_0.17.2      
[17] sjmisc_2.7.6         sjPlot_2.6.1         RColorBrewer_1.1-2   reshape2_1.4.3      
[21] reshape_0.8.8        readr_1.1.1          readxl_1.1.0         mice_3.3.0          
[25] Hmisc_4.1-1          Formula_1.2-3        survival_2.41-3      gridExtra_2.3       
[29] Gmisc_1.7            htmlTable_1.12       Rcpp_1.0.0           dplyr_0.7.8         
[33] cowplot_0.9.3        circlize_0.4.5       caret_6.0-84         ggplot2_3.0.0       
[37] lattice_0.20-35     

loaded via a namespace (and not attached):
  [1] backports_1.1.2     igraph_1.2.2        plyr_1.8.4          lazyeval_0.2.1     
  [5] adabag_4.2          TMB_1.7.14          splines_3.5.0       TH.data_1.0-9      
  [9] digest_0.6.15       foreach_1.4.4       htmltools_0.3.6     gdata_2.18.0       
 [13] magrittr_1.5        checkmate_1.8.5     doParallel_1.0.14   cluster_2.0.7-1    
 [17] recipes_0.1.4       modelr_0.1.2        gower_0.1.2         sandwich_2.5-0     
 [21] strucchange_1.5-1   colorspace_1.3-2    haven_2.1.0         pan_1.6            
 [25] crayon_1.3.4        lme4_1.1-18-1       bindr_0.1.1         zoo_1.8-4          
 [29] iterators_1.0.10    glue_1.3.0          gtable_0.2.0        ipred_0.9-7        
 [33] emmeans_1.2.4       kernlab_0.9-27      shape_1.4.4         jomo_2.6-4         
 [37] abind_1.4-5         scales_0.5.0        mvtnorm_1.0-8       ggeffects_0.6.0    
 [41] plotrix_3.7-4       xtable_1.8-3        Cubist_0.2.2        foreign_0.8-70     
 [45] stats4_3.5.0        prediction_0.3.6    lava_1.6.3          prodlim_2018.04.18 
 [49] glmnet_2.0-16       htmlwidgets_1.2     acepack_1.4.1       modeltools_0.2-22  
 [53] pkgconfig_2.0.1     XML_3.98-1.16       tidyselect_0.2.5    rlang_0.3.0.1      
 [57] munsell_0.4.3       cellranger_1.1.0    tools_3.5.0         party_1.3-1        
 [61] generics_0.0.2      pls_2.7-0           sjlabelled_1.0.14   broom_0.5.0        
 [65] ggridges_0.5.1      evaluate_0.10.1     stringr_1.3.1       kknn_1.3.1         
 [69] yaml_2.1.19         ModelMetrics_1.2.0  knitr_1.20          caTools_1.17.1.1   
 [73] randomForest_4.6-14 forestplot_1.7.2    purrr_0.2.5         mitml_0.3-6        
 [77] coin_1.2-2          nlme_3.1-137        compiler_3.5.0      bayesplot_1.6.0    
 [81] rstudioapi_0.7      e1071_1.7-0         tibble_1.4.2        stringi_1.1.7      
 [85] forcats_0.3.0       Matrix_1.2-14       psych_1.8.4         nloptr_1.2.1       
 [89] stringdist_0.9.5.1  pillar_1.2.2        pwr_1.2-2           GlobalOptions_0.1.0
 [93] estimability_1.3    data.table_1.11.4   bitops_1.0-6        R6_2.2.2           
 [97] latticeExtra_0.6-28 KernSmooth_2.23-15  codetools_0.2-15    MASS_7.3-49        
[101] gtools_3.8.1        assertthat_0.2.0    rprojroot_1.3-2     withr_2.1.2        
[105] mnormt_1.5-5        multcomp_1.4-8      parallel_3.5.0      hms_0.4.2          
[109] rpart_4.1-13        timeDate_3043.102   tidyr_0.8.1         coda_0.19-2        
[113] glmmTMB_0.2.2.0     class_7.3-14        minqa_1.2.4         rmarkdown_1.10     
[117] snakecase_0.9.2     mda_0.4-10          lubridate_1.7.4     base64enc_0.1-3    
```

**Comment by KomodoMatta** [comment]
One thing I forgot to mention is that after running the ``rfe`` function for the first time, I get a warning as well that is likely pertinent:
````
In train.default(x, y, ...) :
  The metric "Accuracy" was not in the result set. ROC will be used instead.
````

``svmrfe <- rfe(...)`` will still run, but I get that warning. 

Thank you again in advance

**Comment by topepo** [comment]
There isn't much that can be done about this. SVMs do not naturally estimate class probabilities so they create a secondary model (basically logistic regression) to convert the decision values to probabilities. This is called [Platt scaling](https://en.wikipedia.org/wiki/Platt_scaling). The problem is that this model can fail and then the SVM model cannot make probability predictions. 

In `kernlab`, IIRC, this process uses some random sampling and there is no way to control the seed. You might try changing the seed before calling `train` or `rfe` to see if that helps.
